### PR TITLE
Allure: add issue and tms links patterns

### LINKED ISSUE #5098: Jira-server-url config to make testId and issueId referencing to project jira in Allure
[//]: # (
. Note: for support questions, please use Stackoverflow or Gitter**. 
. This repository's issues are reserved for feature requests and bug reports.
.
. In case of any problems with Allure Jenkins plugin** please use the following repository 
. to create an issue: https://github.com/jenkinsci/allure-plugin/issues
.
. Make sure you have a clear name for your issue. The name should start with a capital 
. letter and no dot is required in the end of the sentence. An example of good issue names:
.
. - The report is broken in IE11
. - Add an ability to disable default plugins
. - Support emoji in test descriptions
)

#### I'm submitting a ... 
  - [ ] bug report
  - [x] feature request
  - [ ] support request => Please do not submit support request here, see note at the top of this template.

#### What is the current behavior?
Allure generates report where **testID** has `{allure-report-path}/{jira-id}` link. 
An error occurs by click on a link.

#### What is the expected behavior?
Allure generates report where **testID/issueID** has` {jira-server-path}/{jira-id}` link. Alike **[allure.properties](https://docs.qameta.io/allure/#_configuration)** file in JAVA.

image

#### What is the motivation / use case for changing the behavior?
Team members will have possibilities navigate to mentioned Jira tests by click.

#### Please tell us about your environment:

| Allure version | 2.7.0 |
| --- | --- |
| Test framework | [wdio](https://webdriver.io/docs/allure-reporter.html)@latest |
| Allure adaptor | @wdio/allure-reporter@latest |
| Generate report using | allure-teamcity@latest |

#### Other information 

[//]: # (
. e.g. detailed explanation, stacktraces, related issues, suggestions 
. how to fix, links for us to have more context, eg. Stackoverflow, Gitter etc
)

### PR DESCRIPTION
**Pull request by BorisOsipov**
## Proposed changes

Resolve #5098 

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

File: packages/wdio-allure-reporter/src/utils.js [hunk 1]
```diff
@@ -70,3 +70,20 @@ export const getErrorFromFailedTest = (test) => {
...
+
+/**
+ * Substitute task id to link template
+ * @param {string} template - link template
+ * @param {string} id - task id
+ * @returns {string} - link after substitution
+ * @private
+ */
+export const getLinkByTemplate = (template, id) => {
+    if (!template || typeof template !== 'string') {
```
Comment (christian-bromann) [inline packages/wdio-allure-reporter/src/utils.js]: checking of not being a string is enough, no?
```suggestion
    if (typeof template !== 'string') {
```

**Comment by BorisOsipov** [comment]
@christian-bromann fixed

**Comment by BorisOsipov** [comment]
backported manually

**Comment by christian-bromann** [comment]
Thanks @BorisOsipov , released in v5.22.0

**Pull request merged**